src/boot/chid.h

```diff
@@ -45,20 +55,40 @@ assert_cc(offsetof(Device, descriptor) == 0);
 assert_cc(offsetof(Device, chid) == 4);
 assert_cc(offsetof(Device, devicetree.name_offset) == 20);
 assert_cc(offsetof(Device, devicetree.compatible_offset) == 24);
+assert_cc(offsetof(Device, fw.name_offset) == 20);
+assert_cc(offsetof(Device, fw.compatible_offset) == 24);
 assert_cc(sizeof(Device) == 28);
 
 static inline const char* device_get_name(const void *base, const Device *device) {
-        if (device->descriptor != DEVICE_DESCRIPTOR_DEVICETREE)
+        uint32_t off = 0;
+        switch (DEVICE_TYPE_FROM_DESCRIPTOR(device->descriptor))
+        {
```

Comment:
please follow coding style, i.e. place `{` on end of previous line

Reply:
Fixed already in https://github.com/systemd/systemd/pull/35747